src/components/Card.js

```diff
@@ -1,29 +1,54 @@
 export default class Card {
-  constructor(data, cardSelector, handleImageClick) {
-    this._name = data.name;
-    this._link = data.link;
+  constructor(data, cardSelector, handleImageClick, handleDeleteClick,
+    handleLikeClick) {
+    this.name = data.name;
+    this.link = data.link; 
+    this._id = data._id; 
+    this._isLiked = data.isLiked;
     this._cardSelector = cardSelector;
     this._handleImageClick = handleImageClick;
+    this._handleLikeClick = handleLikeClick;
+    this._handleDeleteClick = handleDeleteClick;
+  } 
+
+  getCardId() {
+    return this._id;
   }
 
+  getIsLiked() {
+    return this._isLiked;
+  }
+  
   _setEventListeners() {
     this._likeButton.addEventListener("click", () => {
-      this._handleLikeIcon();
+      this._handleLikeClick(this);
     });
+
     this._deleteButton.addEventListener("click", () => {
-      this._handleDeleteCard();
+      this.handleDeleteCard(this);
```

Comment:
Needs correction: here you should use the provided handler ` this._handleDeleteClick`